core/src/main/java/io/aiven/klaw/service/ApprovalService.java

```diff
@@ -0,0 +1,305 @@
+package io.aiven.klaw.service;
+
+import io.aiven.klaw.config.ManageDatabase;
+import io.aiven.klaw.dao.Approval;
+import io.aiven.klaw.dao.Team;
+import io.aiven.klaw.dao.UserInfo;
+import io.aiven.klaw.error.KlawException;
+import io.aiven.klaw.helpers.HandleDbRequests;
+import io.aiven.klaw.model.enums.ApprovalType;
+import io.aiven.klaw.model.enums.MailType;
+import io.aiven.klaw.model.enums.RequestEntityType;
+import io.aiven.klaw.model.enums.RequestOperationType;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class ApprovalService {
+
+  private final Map<String, List<Approval>> topicApprovals;
+
+  private final Map<String, List<Approval>> aclApprovals;
+
+  private final Map<String, List<Approval>> schemaApprovals;
+
+  private final Map<String, List<Approval>> connectApprovals;
+
+  @Autowired private MailUtils mailService;
+  @Autowired private CommonUtilsService commonUtilsService;
+  @Autowired ManageDatabase manageDatabase;
+
+  @Value("${klaw.approvals.multiApproval:false}")
+  private boolean allowMultiApproval;
+
+  public ApprovalService(
+      Map<String, List<Approval>> topicApprovals,
```

Comment:
I see this error on this constructor "Could not autowire. No beans of 'List<Approval>' or 'Map<String, List<Approval>>' types found."

Reply:
Yea , but could it be an IntelliJ issue not able to find this ?